test/integration/hazelcast_controller_test.go

```diff
@@ -80,9 +80,41 @@ var _ = Describe("Hazelcast controller", func() {
 		return fetchedCR
 	}
 
-	Update := func(hz *hazelcastv1alpha1.Hazelcast) {
+	type MutateFn func(*hazelcastv1alpha1.Hazelcast) *hazelcastv1alpha1.Hazelcast
+
+	SetClusterSize := func(size int32) MutateFn {
+		return func(hz *hazelcastv1alpha1.Hazelcast) *hazelcastv1alpha1.Hazelcast {
+			hz.Spec.ClusterSize = size
+			return hz
+		}
+	}
+
+	Update := func(hz *hazelcastv1alpha1.Hazelcast, fns ...MutateFn) {
 		By("updating the CR with specs successfully")
-		Expect(k8sClient.Update(context.Background(), hz)).Should(Succeed())
+		if len(fns) == 0 {
+			Expect(k8sClient.Update(context.Background(), hz)).Should(Succeed())
+		} else {
+			for {
+				cr := &hazelcastv1alpha1.Hazelcast{}
+				Expect(k8sClient.Get(
+					context.Background(),
+					types.NamespacedName{Name: hz.Name, Namespace: hz.Namespace},
+					cr),
+				).Should(Succeed())
+				for i := range fns {
+					cr = fns[i](cr)
+				}
```

Comment:
One more small suggestion:
```suggestion
				for _, f := range fns {
					cr = f(cr)
				}
```